### PR TITLE
Refactor isUserActive

### LINKED ISSUE #81: Use isUserActive check from routines in 'список заявок'

### PR DESCRIPTION
**Pull request by ya7on**
close #81 

# Please make sure all below checkboxes in the Mandatory section have been checked before submitting your PR (also don't forget to pay attention to the Depending on Circumstances section)

## Mandatory

- [x] The commit message is an imperative and starts with a capital letter (for example, `Add something`, `Remove something`, `Fix something`, etc.).
- [x] I made sure I hadn't put a period at the end of the the commit message(s).

## Depending on Circumstances

Some of the items in the section become mandatory if you are a first-time contributor and/or your changes are relevant to the items.

- [ ] Adding a new parameter I didn't forget to reflect it in `README.md`.
- [ ] Modifying a list of constants or imports I didn't brake an alphabetical order.
- [ ] (for first-time contributors) One of the commits in the pull request adds me to the **Acknowledgements** section in `AUTHORS.md`.

### CODE REVIEW

File: src/viva.js [hunk 1]
```diff
@@ -173,10 +173,9 @@ module.exports = async (robot) => {
   }
 
   function checkIfUserExists (robot, username) {
-    const users = robot.brain.data.users
-    const usernames = Object.values(users).map(user => user.name)
+    const users = routines.getAllUsers(robot).map(user => user.name)
 
-    return usernames.indexOf(username) > -1
+    return users.indexOf(username) > -1
   }
```
Comment (polina-popova) [inline src/viva.js]: I guess we should change the name of the function to `checkIfUserActive` and comment why we don't use direct call to `routines.isUserActive`.

File: src/viva.js [hunk 2]
```diff
@@ -986,48 +956,37 @@ module.exports = async (robot) => {
...
-      .map(user => {
-        return routines.doesUserExist(robot, user).then((isExist) => {
-          return { user, isExist }
-        })
-      })
-
-    const filteredUser = (await Promise.all(users)
-      .then(array => array.filter(item => item.isExist)))
+    const user = (await routines.getAllUsers(robot))
       .find(user => user.user.name === username)
```
Comment (polina-popova) [inline src/viva.js]: Use `routines.findUserByName` instead of repeating search.

File: src/viva.js [hunk 3]
```diff
@@ -986,48 +956,37 @@ module.exports = async (robot) => {
...
+    if (user) {
       let day, month, year
       const dates = []
 
       if (leaveStart !== '*') {
         day = parseInt(leaveStart.split('.')[0])
         month = parseInt(leaveStart.split('.')[1])
         year = parseInt(leaveStart.split('.')[2])
-        find.vivaLasVegas.leaveStart = { day, month, year }
+        user.vivaLasVegas.leaveStart = { day, month, year }
```
Comment (polina-popova) [inline src/viva.js]: Here the changes are only in variable name. I guess it should go to PR with refactoring.

Comment (BehindLoader) [reply src/viva.js]: I change the variable name because its content was changed.

**Comment by eugulixes** [comment]
Closing because it's not relevant.

**Pull request closed without merging**